sonar-ldap-plugin/src/main/java/org/sonar/plugins/ldap/LdapContextFactory.java

```diff
@@ -95,6 +105,30 @@ private InitialDirContext createInitialDirContext(String principal, String crede
     return new InitialLdapContext(getEnvironment(principal, credentials, pooling), null);
   }
 
+  private InitialDirContext createInitialDirContextUsingGssapi(String principal, String credentials) throws NamingException {
+    Configuration.setConfiguration(new Krb5LoginConfiguration());
+    InitialDirContext initialDirContext;
+    try {
+      LoginContext lc = new LoginContext(getClass().getName(), new CallbackHandlerImpl(principal, credentials));
+      lc.login();
+      initialDirContext = Subject.doAs(lc.getSubject(), new PrivilegedExceptionAction<InitialDirContext>() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1604)